engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/JobExecutionEngine.scala

```diff
@@ -78,6 +80,82 @@ final class JobExecutionEngine(
   private lazy val logger: TruffleLogger =
     runtimeContext.executionService.getLogger
 
+  // Independent Runnable that has a list of jobs that should finish within a pre-determined window
+  // and, if not, are interrupted.
+  private class ForceJobCancellations(val pendingJobs: Seq[(Long, RunningJob)])
+      extends Runnable {
+    private val forceInterruptTimeout: Long = 50 * 1000
+
+    override def run(): Unit = {
+      pendingJobs.sortBy(_._1).foreach {
+        case (timeRequestedToCancel, runningJob) =>
+          try {
+            val now                        = System.currentTimeMillis()
+            val timeSinceRequestedToCancel = now - timeRequestedToCancel
+            assert(timeSinceRequestedToCancel > 0)
```

Comment:
There is `assertInJVM` that is preferred to use when honoring `-ea` is the goal.